src/sfdt/toggleBookmark.ts

```diff
@@ -5,6 +5,88 @@ import {
 	isBookmarkStart,
 	isBookmarkEnd,
 } from '../queryBookmark'
+import Stack from '../stack'
+
+const makeToggleOff = (inlines, name) => {
+  const newInlines = []; // Should act as queue for getting new list of inlines
+  const stackOfBookmarks = new Stack(); // will be used to find relationship of bookmarks
+  let toggleOffAtIndex = null;
+
+  inlines.forEach((inline, index) => {
```

Comment:
note: the previous function had a lot of comments to explain what was going on. please also add comments in your new one to help the next reader. logic blocks like this are hard to follow otherwise